routes/user.playlist.js

```diff
@@ -252,139 +255,210 @@ router.route('/:playlist_name/export3')
-                                pObj = slide.addText(songObj.songs[x].lyric[i][j], {
-                                    x: 'c', //x position
-                                    y: margin, //y position
-                                    cx: '100%', //width
-                                    cy: 40,
-                                    font_size: 40,
-                                    align: 'center',
-                                    color: {
-                                        type: 'solid',
-                                        color: 'ffffff'
-                                    }
-                                });
-                                margin += 50;
-                            }
-                            margin += 150
+                            color: 'ffffff'
                         }
-
-                    }
-                })
-                callback();
+                    });
+                    margin += 50;
+                }
+                margin += 150
             }
-
-
-            function finalize() {
-                var out = fs.createWriteStream(filename + '.pptx');
-                out.on('error', function(err) {
-                    console.log(err);
-                });
-                res.writeHead(200, {
-                    "Content-Type": "application/vnd.openxmlformats-officedocument.presentationml.presentation",
-                    'Content-disposition': 'attachment; filename=' + filename + '.pptx'
-                });
-                pptx.generate(res);
+        }
+        var out = fs.createWriteStream(filename + '.pptx');
+        pptx.generate(out, {
+            'finalize': function(written) {
+                console.log('Finish to create a PowerPoint file.\nTotal bytes created: ' + written + '\n');
+                files.push(filename)
+                if (index === arr.length - 1) {
+                    setTimeout(() => {
+                        var zip = archiver('zip')
+                        files.forEach((file) => {
+                            zip.file(file + '.pptx')
+                        })
+                        zip.finalize()
+                        res.writeHead(200, {
+                            'Content-Type': 'application/zip',
+                            'Content-disposition': 'attachment; filename=' + playlistName + '.zip'
+                        });
+                        zip.pipe(res)
+                    }, 1000);
+                }
+            },
+            'error': function(err) {
+                console.log(err);
             }
+        });
+    })
+}
 
-            async.series([generateSlides], finalize);
+//this function map the original 2darray to be 2d array of exactly two songs
+//takes the original 2darray and return the new mapped 2darray
+function create2dArrayOfTwoSongs(songs2d) {
+    var newSongsArr = []
+    var temp;
+    songs2d.forEach((songs) => {
```

Comment:
you can just do a return songs2d.map()
that is the same as what's happening here and lets you drop newSongsArr.